integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/column/ExpressionDependencyCollector.java

```diff
@@ -74,7 +103,19 @@ static void collectFromNode(ColumnLevelLineageContext context, LogicalPlan node)
 
     expressions.stream()
         .forEach(
-            expr -> traverseExpression((Expression) expr, expr.exprId(), context.getBuilder()));
+            expr ->
+                traverseExpression(
+                    (Expression) expr,
+                    expr.exprId(),
+                    TransformationInfo.identity(),
+                    context.getBuilder()));
+
+    if (datasetTransformation.isDefined()) {
+      ExprId exprId = NamedExpression.newExprId();
```

Comment:
because things like `Project`, `Aggregate`,`Filter` etc. are not expressions so we can't put them in expressionDependencies for later resolving. By creating them this dummy ExprId we give them unique identifier and we can treat them just like expression that has expression dependencies (which fits neatly with current dependency collection mechanism).